nifi-nar-bundles/nifi-gcp-bundle/nifi-gcp-processors/src/main/java/org/apache/nifi/processors/gcp/drive/FetchGoogleDrive.java

```diff
@@ -80,6 +84,62 @@
+    private static final AllowableValue EXPORT_PDF = new AllowableValue("application/pdf", "PDF");
+    private static final AllowableValue EXPORT_RICH_TEXT = new AllowableValue("application/rtf", "Rich Text");
+    private static final AllowableValue EXPORT_EPUB = new AllowableValue("application/epub+zip", "EPUB");
+
+    // Shared Export Types
+    private static final AllowableValue EXPORT_HTML_DOC = new AllowableValue("application/zip", "Web Page (HTML)");
+    private static final AllowableValue EXPORT_PLAIN_TEXT = new AllowableValue("text/plain", "Plain Text");
+
+    // Google Spreadsheet Export Types
+    private static final AllowableValue EXPORT_MS_EXCEL = new AllowableValue("application/vnd.openxmlformats-officedocument.spreadsheetml.sheet", "Microsoft Excel");
+    private static final AllowableValue EXPORT_OPEN_SPREADSHEET = new AllowableValue("application/x-vnd.oasis.opendocument.spreadsheet", "OpenDocument Spreadsheet");
+    private static final AllowableValue EXPORT_PDF_SPREADSHEET = new AllowableValue("application/pdf", "PDF");
+    private static final AllowableValue EXPORT_CSV = new AllowableValue("text/csv", "CSV (first sheet only)",
+        "Comma-separated values. Only the first sheet will be exported.");
+    private static final AllowableValue EXPORT_TSV = new AllowableValue("text/tab-separated-values", "TSV (first sheet only)",
+        "Tab-separate values. Only the first sheet will be exported.");
+    private static final AllowableValue EXPORT_HTML_SPREADSHEET = new AllowableValue("text/html", "Web Page (HTML)");
+
+    // Google Presentation Export Types
+    private static final AllowableValue EXPORT_MS_POWERPOINT = new AllowableValue("application/vnd.openxmlformats-officedocument.presentationml.presentation", "Microsoft PowerPoint");
+    private static final AllowableValue EXPORT_OPEN_PRESENTATION = new AllowableValue("application/vnd.oasis.opendocument.presentation", "OpenDocument Presentation");
+    private static final AllowableValue EXPORT_PNG = new AllowableValue("image/png", "PNG (first slide only)");
+    private static final AllowableValue EXPORT_JPEG = new AllowableValue("image/jpeg", "JPEG (first slide only)");
+    private static final AllowableValue EXPORT_SVG = new AllowableValue("image/svg+xml", "SVG (first slide only)",
+        "Scalable Vector Graphics. Only the first slide will be exported.");
+
+    // Drawings Export Types
+    private static final AllowableValue EXPORT_PNG_DRAWING = new AllowableValue("image/png", "PNG");
+    private static final AllowableValue EXPORT_JPEG_DRAWING = new AllowableValue("image/jpeg", "JPEG");
+    private static final AllowableValue EXPORT_SVG_DRAWING = new AllowableValue("image/svg+xml", "SVG");
+
+    private static final Map<String, String> fileExtensions = new HashMap<>();
+    static {
+        fileExtensions.put("application/vnd.openxmlformats-officedocument.wordprocessingml.document", ".docx");
```

Comment:
```suggestion
        fileExtensions.put(EXPORT_MS_WORD.getValue(), ".docx");
```

Is it worth using the MIME Type strings from the `AllowableValues`, rather than duplicating the hardcoded String - more likely to ensure consistency and prevent future maintenance headaches

Reply:
Related to this point, it might be more efficient to use an `enum` that implements `DescribedValue` to define the allowed values, and then all properties could be kept together.

Reply:
Thanks @ChrisSamo632 , that's a good call. Pushed an update to handle that.

@exceptionfactory I did look at an `enum` but ultimately didn't go that route simply because several of the values are shared both multiple properties. For example, 3 of the types I think support `text/plain` as well as `pdf`. Two also support HTML and 2 support png/jpeg, etc.

Reply:
Thanks for the reply @markap14, that's a good point, and makes sense on further review.